### PR TITLE
#842: revoke non-origin permissions

### LINKED ISSUE #842: Add support for revoking non-origin optional permissions

### PR DESCRIPTION
**Pull request by twschiller**
Closes #842

### CODE REVIEW

**Review by fregante — COMMENTED** [review]
Screen Shot

I see this in Firefox. Might this be a bug in `getAdditionalPermissions`?

I think it's because, like for contentScript’s origins, the extension do get the permission to access a site even if it's not listed in the `manifest.permissions` key.

`getAdditionalPermissions` doesn't know that.

A quick fix here would be to include that permission in our manifest.

**Review by fregante — COMMENTED** [review]
Tested, it works, but `devtools` should be part of the permissions.

I also re-tested #845 with the `devtools` permission but no luck

**Review by fregante — APPROVED** [review]

**Comment by twschiller** [comment]
> I see this in Firefox

I see it in Chrome too

> A quick fix here would be to include that permission in our manifest

I might just hide it from the Settings page, as the user should never need to revoke it?

**Comment by fregante** [comment]
> the user should never need to revoke it?

I don't think it can be revokable in the first place since it's not part of optionalPermissions. You can use this function to specifically filter the list: https://github.com/pixiebrix/pixiebrix-extension/blob/97be358700d4f1b277ed20f275c18a26a4e6408f/src/utils/permissions.ts#L26

File: src/manifest.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "storage",
     "identity",
     "tabs",
+    "devtools",
```
Comment (twschiller) [inline src/manifest.json:29]: "devtools" is not listed as an available permission here:

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions

Give that the devtools working now without this, I'm going to keep it out of here, especially as I don't want the extension to deactivate on the next update (Chrome deactivates extensions that add new permissions)

**Pull request merged**